src/ClientHandle.cpp

```diff
@@ -2228,7 +2235,10 @@ void cClientHandle::ServerTick(float a_Dt)
 
 	{
 		cCSLock lock(m_CSState);
-		if (m_State == csAuthenticated)
+		if (
+			(m_State == csAuthenticated) &&
+			(cRoot::Get()->GetServer()->GetResourcePackUrl().empty() || (m_ResourcePackStatus != -1))
+		)
```

Comment:
So when does the client start streaming chunks if resource pack url is not empty?

Reply:
I'm starting to realize that this needs a different approach. Not exactly sure how to solve this, but as soon as chunks are streamed, the resource pack prompt closes.

Reply:
Fixed the issue.